sqlglot/dialects/snowflake.py

```diff
@@ -874,7 +874,12 @@ def _parse_location_path(self) -> exp.Var:
             ):
                 parts.append(self._advance_any(ignore_reserved=True))
 
-            return exp.var("".join(part.text for part in parts if part))
+            part_texts = (
+                f'"{part.text}"' if part.token_type == TokenType.IDENTIFIER else part.text
+                for part in parts
+                if part
+            )
+            return exp.var("".join(part_texts))
```

Comment:
note: I'm generally wondering if we should return a single `exp.Var` with the stringified identifier here (which could have a value like`'@"my_DB"."schEMA1"."MYstage"'`), or whether we could potentially return a more structured type, something like the `exp.Table` that gets created when parsing an expression like `SELECT * FROM "my_DB"."schEMA1"."MYstage"`

```
Table(
      this=Identifier(this='MYstage', quoted=True),
      db=Identifier(this='schEMA1', quoted=True),
      catalog=Identifier(this='my_DB', quoted=True))
```

Curious to get your thoughts on this!

Reply:
The note in this [section](https://docs.snowflake.com/en/user-guide/querying-stage#required-parameters) says:

> The URI string for an external storage location (Amazon S3, Google Cloud Storage, or Microsoft Azure) must be enclosed in single quotes; however, you can enclose any URI string in single quotes, which allows special characters, including spaces, in location and file names. For example: `Internal: '@~/path 1/file 1.csv' [...]`

So it seems like it should be fine to concatenate the parts' SQL and produce a string instead of a var. I think this aligns with what you said in the first sentence, but we'd produce a `Literal` string instead of a `Var`.

Reply:
Ok thanks for the feedback @georgesittas - do you want me to update the PR and return a `Literal` instead of a `Var`..?

Reply:
Yep! Thanks